### ISSUE TITLE
Error: unhandled exception: errno: 9 `Bad file descriptor' [IOError]

### ISSUE THREAD

**Issue by retsyo** [nim-lang/Nim#16502]
I am using latest cloned&built nim
```
$ nim
Nim Compiler Version 1.5.1 [Windows: amd64]
Compiled at 2020-12-29
Copyright (c) 2006-2020 by Andreas Rumpf

```

### Example
```nim
echo 1
```

### Current Output

when I compile the `a.nim` with `nim c -r --app:gui a.nim`, I get an error msgbox

```
R:\a.nim(1)
a
E\msys64\home\USER\_nim\nim\lib\system\io.nim(791) echoBinSafe
E:\msys64\home\USER\_nim\nim\lib\system\io.nim(156) checkErr
E:\msys64\home\USER\_niminim\lib\system\io.nim(139) raiseEIO
Error: unhandled exception: errno: 9 `'Bad file descriptor` [IOError]
```

when I compile `a.nim` as `nim c -r a.nim`, I see
```
1
```
as expected

### Expected Output

yes, I know `--app:gui` means to compile a `GUI`  app without console, so I expect `nim c -r --app:gui a.nim` runs but without error msgbox. Why do I need this? Because I write some `echo xx` in my app which are used for debugging. When the app is finished, I need to release a GUI app to the user without console, but I am lazy to remove the `echo xx` statements.

thanks

**Comment by ringabout** [comment]
Workaround:
Use `nim c -r  --app:gui -d:nimLegacyEchoNoRaise`

**Comment by ringabout** [comment]
see https://github.com/nim-lang/Nim/pull/16367

**Comment by timotheecour** [comment]
https://github.com/nim-lang/Nim/pull/16367 was still correct to raise; `-d:nimLegacyEchoNoRaise` is indeed usable here, but you probably should use:
```nim
template myEcho(args: varargs[untyped]): untyped =
  # adapt if needed so echo still renders at CT using `when nimvm`
  when appType == "gui": discard
  else: echo(args)
```

(or, if it's meaningful to support, `echo` at RT with app:gui could be implemented by rendering a window , eg System.Windows.MessageBox https://mcpmag.com/articles/2016/06/09/display-gui-message-boxes-in-powershell.aspx)

**Comment by Clyybber** [comment]
IMO the change should be reverted, what worth is it raising if you can't display the exception (on platforms where we don't have this messagebox)?
It's only useful if you intend to catch the exception, but since all code that relies on that will be new code, the raising version of echo should be a new proc.

**Comment by ringabout** [comment]
> (or, if it's meaningful to support, echo at RT with app:gui could be implemented by rendering a window , eg System.Windows.MessageBox

no

It is easy to fix. Just adding `not defined(guiapp)` is enough without reverting original PR.


```nim
echo 1
```
In windows, I got a messagebox and I can catch this exception.
![image](https://user-images.githubusercontent.com/43030857/103331948-19541e80-4aa3-11eb-8fff-8dd1a6d9e467.png)


Note that `app:gui` is specific to `Windows`
see [disable --app:gui for non Windows targets; fixes #2576](https://github.com/nim-lang/Nim/commit/d270245a0171c4532c9121e7e27dfd422f123840)
